Tests/Common/Background Queue/QueueRunRequestTest.swift

```diff
@@ -286,6 +287,30 @@ class QueueRunRequestIntegrationTest: IntegrationTest {
         XCTAssertTrue(queueStorage.getInventory().isEmpty)
     }
 
+    func test_givenMultipleErrorRequest_expectTasksToBeDeletedFor400BadRequestOnly() {
```

Comment:
Good test to test what happens when a 400 occurs for a task at the start of a group. 

We also need to have a test that tests when a 400 happens and the task is *not* the start of a group.